homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,148 @@
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement that is used."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return the current HVAC mode."""
+        return self._hvac_mode
+
+    @property
+    def hvac_action(self):
+        """Return the current HVAC action."""
+        return self._hvac_action
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        return [HVAC_MODE_HEAT, HVAC_MODE_OFF, HVAC_MODE_AUTO]
+
+    @callback
+    def update_data(self):
+        """Update data."""
+        if self._coordinator.last_update_success:
+            state = self._coordinator.data
+            if state["power"]:
+                if state["auto_mode"]:
+                    self._hvac_mode = HVAC_MODE_AUTO
+                else:
+                    self._hvac_mode = HVAC_MODE_HEAT
+
+                if state["active"]:
+                    self._hvac_action = CURRENT_HVAC_HEAT
+                else:
+                    self._hvac_action = CURRENT_HVAC_IDLE
+            else:
+                self._hvac_mode = HVAC_MODE_OFF
+                self._hvac_action = CURRENT_HVAC_OFF
+
+            self._current_temperature = state["room_temp"]
+            self._target_temperature = state["thermostat_temp"]
+
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self):
+        """Call when the climate device is added to hass."""
+        state = await self.async_get_last_state()
+        if state is not None:
+            self._hvac_mode = state.state
+            self._hvac_action = state.attributes[ATTR_HVAC_ACTION]
+            self._current_temperature = state.attributes[ATTR_CURRENT_TEMPERATURE]
+            self._target_temperature = state.attributes[ATTR_TEMPERATURE]
+
+        self.async_on_remove(self._coordinator.async_add_listener(self.update_data))
+
+    async def async_update(self):
+        """Update the climate entity."""
+        await self._coordinator.async_request_refresh()
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        device = self._device
+
+        if hvac_mode == HVAC_MODE_OFF:
+            await device.async_request(device.api.set_power, 0)
+
+        elif hvac_mode == HVAC_MODE_AUTO:
+            await device.async_request(device.api.set_power, 1)
+            await device.async_request(device.api.set_mode, 1, 0)
+
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await device.async_request(device.api.set_power, 1)
+            await device.async_request(device.api.set_mode, 0, 0)
```

Comment:
We need to check if the operation was successful and update the state attributes if so. See light.py and switch.py.

Reply:
I'am not 100% sure that I know what you mean.
```
self._attr_is_on = False
```
This? I can't see a clear pattern in switch and light.

Reply:
@felipediel can you please elaborate?

Reply:
If the request worked, we want to update the state instantly, in the end of the operation, writing the state to the state machine, instead of waiting for updates from the data update coordinator.

Reply:
Example:
https://github.com/home-assistant/core/blob/513d6cc4676bf93c813a3c230429eed14a40b386/homeassistant/components/broadlink/switch.py#L148-L152

Reply:
We need to write all the state attributes we are changing, not only _attr_is_on.

Reply:
Here is a list of attributes we can use:
https://github.com/home-assistant/core/blob/513d6cc4676bf93c813a3c230429eed14a40b386/homeassistant/components/climate/__init__.py#L178-L205

Reply:
@felipediel I finally found the time to pick up work. I added the update calls and used _attr almost everywhere. Please have a look :) I tested everything and it works flawless.

Reply:
@felipediel can you please have a look again :)

Reply:
@felipediel can you please read the code?